sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -159,6 +156,10 @@ trait CodegenSupport extends SparkPlan {
           BoundReference(i, attr.dataType, attr.nullable).genCode(ctx)
         }
       }
+    } match {
```

Comment:
I think this construction is confusing. Can this be split into two statements rather than chain a block and match?